File: langs/scala/Prime.scala
object Prime {
   def sieve(size: Int): Array[Boolean] = {
      if (size <= 1) return Array.fill(size)(false)
      val sieveArray = Array.fill(size)(true)
      sieveArray(0) = false
      sieveArray(1) = false
      val root = math.sqrt(size).toInt + 1
      for (i <- 2 until root) {
         if (sieveArray(i)) {
            for (j <- (i * i) until sieveArray.length by i) {
               sieveArray(j) = false
            }
         }
      }
      return sieveArray
   }
   
   def countPrime(n: Int): Int = {
      return sieve(n + 1).count(x => x)
   }
   
   def main(args: Array[String]) = {
      var n = if (args.length >= 1) args(0).toInt else 10000000
      println(countPrime(n))
   }
}

File: langs/clojure/Prime.clj
(ns Prime (:gen-class))

(defn sieve [size]
    (if (<= size 1)
        (boolean-array size false)
        (let [sieveArray (boolean-array size true)
              root (+ 1 (int (Math/sqrt size)))]
            (aset-boolean sieveArray 0 false)
            (aset-boolean sieveArray 1 false)
            (doseq [i (range 2 root)]
                (when (aget sieveArray i)
                    (doseq [j (range (* i i) size i)]
                        (aset-boolean sieveArray j false))))
            sieveArray)))

(defn countPrime [n]
    (count (filter identity (sieve (+ n 1)))))

(defn -main [& args]
    (let [n (if (not-empty args) (Integer/parseInt (first args)) 10000000)]
        (println (countPrime n))))